src/integrationTest/java/integration/tests/ConnectionTest.java

```diff
@@ -99,8 +99,8 @@ void connect(boolean useDatabase, boolean useEngine) throws SQLException {
         String engine = useEngine ? params.getEngine() : null;
         ConnectionInfo updated = new ConnectionInfo(params.getPrincipal(), params.getSecret(), params.getEnv(), database, params.getAccount(), engine, params.getApi());
         String url = updated.toJdbcUrl();
-        String query = "SELECT TOP 1 * FROM information_schema.tables";
-        boolean expectedSuccess = updated.getApi() != null || infraVersion == 2 ? useDatabase : useDatabase || useEngine;
+        String query = "SELECT version()";
+        boolean expectedSuccess = updated.getApi() == null || useDatabase;
```

Comment:
Isn't this test supposed to work even if `useDatabase` is false? 
In general, let's avoid adding complicated if logic like we had in this test before. It only leads to confusion on what the test is testing.

Reply:
Agree. Will be fixed

Reply:
Fixed